### PR TITLE
Fixed homework submissions functionality

### PR DESCRIPTION
**Pull request by bradweee**
### Changelog
- Fixed functionality to add homework submissions
- Edited db to index homework submissions on course session participants
- Reviewed w/ Muffy & Howard :) 

### To-Dos
- Remove the hardcoded course session id
- Add view to see all assignments
- Add view to see submissions per student

### Checklist:
- [x]  My code follows the style guideline
- [x]  I have performed a self-review of my own code

### CODE REVIEW

File: app/controllers/application_controller.rb [hunk 1]
```diff
@@ -11,6 +11,10 @@ def logged_in?
     !current_user.nil?
   end
 
+  def set_student
+    @student = CourseSessionParticipant.student.find_or_initialize_by(user: current_user, course_session_id: 2) #@course_session)
+  end
```
Comment (shaolinhoward) [inline app/controllers/application_controller.rb:16]: need to unhardcode

File: config/routes.rb [hunk 2]
```diff
@@ -1,9 +1,11 @@
 Rails.application.routes.draw do
 
-  resources :homework_submissions
+  resources :homework_submissions
```
Comment (shaolinhoward) [inline config/routes.rb:3]: lets clean up this trailing space

File: db/schema.rb [hunk 3]
```diff
@@ -46,6 +46,19 @@
     t.index ["user_id"], name: "index_course_session_participants_on_user_id"
   end
 
+  create_table "course_session_users", force: :cascade do |t|
```
Comment (shaolinhoward) [inline db/schema.rb:49]: I think we need to remove this

File: db/schema.rb [hunk 4]
```diff
@@ -55,19 +68,19 @@
...
 
   create_table "homework_submissions", force: :cascade do |t|
-    t.bigint "user_id", null: false
     t.bigint "homework_id", null: false
     t.string "pull_request"
     t.boolean "is_public"
     t.datetime "created_at", precision: 6, null: false
-    t.datetime "updated_at", precision: 6, null: false 
+    t.datetime "updated_at", precision: 6, null: false
+    t.bigint "course_session_participant"
```
Comment (shaolinhoward) [inline db/schema.rb:81]: this too

File: db/schema.rb [hunk 5]
```diff
@@ -55,19 +68,19 @@
     t.datetime "updated_at", precision: 6, null: false
     t.string "repository_link"
     t.bigint "lectures_id"
-    t.integer "status", default: 0, null: false
+    t.integer "status"
```
Comment (shaolinhoward) [inline db/schema.rb:71]: did we want to change this?

File: db/schema.rb [hunk 6]
```diff
@@ -91,7 +104,7 @@
     t.datetime "updated_at", precision: 6, null: false
     t.bigint "course_session_id"
     t.datetime "schedule_date"
-    t.integer "topic", default: 0, null: false
+    t.integer "topic"
```
Comment (shaolinhoward) [inline db/schema.rb:107]: hrm, maybe put this back?

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (bradweee) [reply app/controllers/application_controller.rb:16]: This is what Muffy is going to work on tomorrow. Should I unhardcode before the merge or leave as is?

File: db/schema.rb (hunk 3 above)
Comment (bradweee) [reply db/schema.rb:49]: Removed

File: db/schema.rb (hunk 4 above)
Comment (bradweee) [reply db/schema.rb:81]: Removed

File: db/schema.rb (hunk 6 above)
Comment (bradweee) [reply db/schema.rb:107]: Put back

File: db/schema.rb (hunk 5 above)
Comment (bradweee) [reply db/schema.rb:71]: Put back

File: db/migrate/20200707064558_drop_course_session_users.rb [hunk 7]
```diff
@@ -0,0 +1,4 @@
+class DropCourseSessionUsers < ActiveRecord::Migration[6.0]
+  def change
```
Comment (shaolinhoward) [inline db/migrate/20200707064558_drop_course_session_users.rb:2]: oops empty migration

Comment (bradweee) [reply db/migrate/20200707064558_drop_course_session_users.rb:2]: updated

Comment (bradweee) [reply db/migrate/20200707064558_drop_course_session_users.rb:2]: Added in content

File: db/schema.rb (hunk 4 above)
Comment (shaolinhoward) [reply db/schema.rb:81]: it came back after migration. I think we need to clean this up somewhere.. I think lets get this PR in and handle it tomorrow

File: app/controllers/application_controller.rb (hunk 1 above)
Comment (muffy) [reply app/controllers/application_controller.rb:16]: I'll take care of it.

**Pull request merged**